File: Daily/2379/2379.java
class Solution {
    public int minimumRecolors(String blocks, int k) {

        int  n = blocks.length();
        int minRecolors = Integer.MAX_VALUE;

        for (int i = 0; i <= n - k; i++) {
            int currentRecolors = 0;

            for (int j = i; j < i + k; j++) {
                if (blocks.charAt(j) == 'W') {
                    currentRecolors++;
                }
            }

            minRecolors = Math.min(minRecolors, currentRecolors);
        }
        return minRecolors;
    }
}

File: Daily/2379/2379.py
class Solution:
    def minRecolors(self, blocks: str, k: int) -> int:
        n = len(blocks)
        min_recolors = float('inf')
        
        current_recolors = 0
        for i in range(k):
            if blocks[i] == 'W':
                current_recolors += 1
        min_recolors = current_recolors
        
        for i in range(k, n):
            if blocks[i - k] == 'W':
                current_recolors -= 1
            if blocks[i] == 'W':
                current_recolors += 1
            min_recolors = min(min_recolors, current_recolors)
            
        return min_recolors